### ISSUE TITLE
Panic when querying an M:N model after inserting/updating records

### ISSUE THREAD

**Issue by TKTheTechie** [prisma/prisma#5212]
## Bug description

If you have a simple m:n schema, and try to add to an existing relationship, a findMany() query will throw a panic.

## Schema
```
model Post {
  author     String @Id
  lastUpdated DateTime   @default(now())
  categories Category[]
}

model Category {
  id     Int     @id 
  posts  Post[]
}
```

Run the following code:
```
const categories = [
  { create: { id: 1 }, where: { id: 1 } },
  { create: { id: 2 }, where: { id: 2 } },
]
await db.post.upsert({
  where: { author: 'author' },
  create: {
    author: 'author',
    categories: {
      connectOrCreate: categories,
    },
  },
  update: {
    categories: { connectOrCreate: categories },
  },
});


await db.post.upsert({
  where: { author: 'author2' },
  create: {
    author: 'author2',
    categories: {
      connectOrCreate: categories,
    },
  },
  update: {
    categories: { connectOrCreate: categories },
  },
})

 const posts= await prisma.post.findMany({
  include:{
   categories: true
  }
 }); // Throws a panic
```
## Prisma information

```
"@prisma/client": "^2.15.0-dev.89",
"@prisma/cli": "^2.15.0-dev.89"
```

## Environment & setup

- OS: Windows
- Database: SQLite
- Node.js version: v14.11.0

## Error
```
(node:25504) UnhandledPromiseRejectionWarning: Error: 
Invalid `prisma.post.findMany()` invocation:



  PANIC in query-engine\core\src\interpreter\query_interpreters\nested_read.rs:78:56
1

This is a non-recoverable error which probably happens when the Prisma Query Engine has a panic.

https://github.com/prisma/prisma/issues/new?body=Hi+Prisma+Team%21+My+Prisma+Client+just+crashed.+This+is+the+report%3A%0A%23%23+Versions%0A%0A%7C+Name++++++++++++%7C+Version++++++++++++%7C%0A%7C-----------------%7C--------------------%7C%0A%7C+Node++++++++++++%7C+v14.11.0+++++++++++%7C+%0A%7C+OS++++++++++++++%7C+windows++++++++++++%7C%0A%7C+Prisma+Client+++%7C+2.15.0-dev.89++++++%7C%0A%7C+Query+Engine++++%7C+query-engine+e51dc3b5a9ee790a07104bec1c9477d51740fe54%7C%0A%7C+Database++++++++%7C+undefined%7C%0A%0A%0A%0A%23%23+Query%0A%60%60%60%0Aquery+%7B%0A++findManyEventMeshClients+%7B%0A++++clientId%0A++++nodeName%0A++++userName%0A++++createdAt%0A++++lastUpdated%0A++++subscriptions+%7B%0A++++++subscription%0A++++%7D%0A++%7D%0A%7D%0A%60%60%60%0A%0A%23%23+Logs%0A%60%60%60%0A%3A%2F%2F127.0.0.1%3A63435++%0A++engine+Search+for+Query+Engine+in+C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C.prisma%5Cclient++%0A++engine+stdout++PANIC+in+query-engine%5Ccore%5Csrc%5Cinterpreter%5Cquery_interpreters%5Cnested_read.rs%3A78%3A56%0A1++%0A++engine+TypeError%3A+this.currentRequestPromise.cancel+is+not+a+function%0A++engine+++++at+NodeEngine.handlePanic+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A26519%3A36%29%0A++engine+++++at+NodeEngine.setError+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A26465%3A16%29%0A++engine+++++at+LineStream.%3Canonymous%3E+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A26760%3A24%29%0A++engine+++++at+LineStream.emit+%28events.js%3A314%3A20%29%0A++engine+++++at+addChunk+%28_stream_readable.js%3A307%3A12%29%0A++engine+++++at+readableAddChunk+%28_stream_readable.js%3A282%3A9%29%0A++engine+++++at+LineStream.Readable.push+%28_stream_readable.js%3A221%3A10%29%0A++engine+++++at+LineStream.Transform.push+%28_stream_transform.js%3A166%3A32%29%0A++engine+++++at+LineStream._pushBuffer+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A23721%3A19%29%0A++engine+++++at+LineStream._transform+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A23715%3A10%29++%7B%22timestamp%22%3A%22Jan+20+10%3A48%3A40.783%22%2C%22level%22%3A%22ERROR%22%2C%22fields%22%3A%7B%22message%22%3A%22PANIC%22%2C%22reason%22%3A%221%22%2C%22file%22%3A%22query-engine%5C%5Ccore%5C%5Csrc%5C%5Cinterpreter%5C%5Cquery_interpreters%5C%5Cnested_read.rs%22%2C%22line%22%3A78%2C%22column%22%3A56%7D%2C%22target%22%3A%22query_engine%22%7D++%0A++engine+%7B%0A++engine+++error%3A+Error%3A+read+ECONNRESET%0A++engine+++++++at+TCP.onStreamRead+%28internal%2Fstream_base_commons.js%3A209%3A20%29+%7B%0A++engine+++++errno%3A+-4077%2C%0A++engine+++++code%3A+%27ECONNRESET%27%2C%0A++engine+++++syscall%3A+%27read%27%0A++engine+++%7D%0A++engine+%7D++%0A++engine+%7B%0A++engine+++error%3A+Error%3A+connect+ECONNREFUSED+127.0.0.1%3A63435%0A++engine+++++++at+TCPConnectWrap.afterConnect+%5Bas+oncomplete%5D+%28net.js%3A1145%3A16%29+%7B%0A++engine+++++errno%3A+-4078%2C%0A++engine+++++code%3A+%27ECONNREFUSED%27%2C%0A++engine+++++syscall%3A+%27connect%27%2C%0A++engine+++++address%3A+%27127.0.0.1%27%2C%0A++engine+++++port%3A+63435%0A++engine+++%7D%0A++engine+%7D++%0A++engine+There+is+a+child+that+still+runs+and+we+want+to+start+again++%0A++engine+%7B+cwd%3A+%27C%3A%5C%5Cdev%5C%5Cevent-mesh-graphql-server%5C%5Cprisma%27+%7D++%0A++engine+Search+for+Query+Engine+in+C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C.prisma%5Cclient++%0A++engine+%7B+flags%3A+%5B+%27--enable-raw-queries%27+%5D+%7D++%0A++engine+port%3A+63438++%0A++engine+Client+Version%3A+2.15.0-dev.89++%0A++engine+Engine+Version%3A+query-engine+e51dc3b5a9ee790a07104bec1c9477d51740fe54++%0A++engine+Active+provider%3A+sqlite++%0A++engine+stdout++Starting+a+sqlite+pool+with+9+connections.++%0A++engine+stdout++Started+http+server+on+http%3A%2F%2F127.0.0.1%3A63438++%0A++engine+Search+for+Query+Engine+in+C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C.prisma%5Cclient++%0A++engine+stdout++PANIC+in+query-engine%5Ccore%5Csrc%5Cinterpreter%5Cquery_interpreters%5Cnested_read.rs%3A78%3A56%0A1++%0A++engine+TypeError%3A+this.currentRequestPromise.cancel+is+not+a+function%0A++engine+++++at+NodeEngine.handlePanic+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A26519%3A36%29%0A++engine+++++at+NodeEngine.setError+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A26465%3A16%29%0A++engine+++++at+LineStream.%3Canonymous%3E+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A26760%3A24%29%0A++engine+++++at+LineStream.emit+%28events.js%3A314%3A20%29%0A++engine+++++at+addChunk+%28_stream_readable.js%3A307%3A12%29%0A++engine+++++at+readableAddChunk+%28_stream_readable.js%3A282%3A9%29%0A++engine+++++at+LineStream.Readable.push+%28_stream_readable.js%3A221%3A10%29%0A++engine+++++at+LineStream.Transform.push+%28_stream_transform.js%3A166%3A32%29%0A++engine+++++at+LineStream._pushBuffer+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A23721%3A19%29%0A++engine+++++at+LineStream._transform+%28C%3A%5Cdev%5Cevent-mesh-graphql-server%5Cnode_modules%5C%40prisma%5Cclient%5Cruntime%5Cindex.js%3A23715%3A10%29++%7B%22timestamp%22%3A%22Jan+20+10%3A48%3A41.036%22%2C%22level%22%3A%22ERROR%22%2C%22fields%22%3A%7B%22message%22%3A%22PANIC%22%2C%22reason%22%3A%221%22%2C%22file%22%3A%22query-engine%5C%5Ccore%5C%5Csrc%5C%5Cinterpreter%5C%5Cquery_interpreters%5C%5Cnested_re%3A+Error%3A+read+ECONNRESET%0A++engine+++++++at+TCP.onStreamRead+%28internal%2Fstream_base_commons.js%3A209%3A20%29+%7B%0A+7D%0A++engine+%7D++%0A%60%60%60%0A%0A%23%23+Client+Snippet%0A%60%60%60ts%0A%2F%2F+PLEASE+FILL+YOUR+CODE+SNIPPET+HERE%0A%60%60%60%0A%0A%23%23+Schema%0A%60%60%60prisma%0A%2F%2F+PLEASE+ADD+YOUR+SCHEMA+HERE+IF+POSSIBLE%0A%60%60%60%0A&title=PANIC+in+query-engine%5Ccore%5Csrc%5Cinterpreter%5Cquery_interpreters%5Cnested_read.rs%3A78%3A56%0A1&template=bug_report.md
This is a non-recoverable error which probably happens when the Prisma Query Engine has a panic.
```
## Resolution/WorkAround

Add an additional field to the Category table:

```
lastUpdated   DateTime                 @updatedAt
```

**Comment by pantharshit00** [comment]
Thanks for the reproduction, I can reproduce this. 

I have posted a reproduction here: https://github.com/harshit-test-org/prisma-issue-5212

**Comment by Weakky** [comment]
Hey 👋, this was fixed by https://github.com/prisma/prisma-engines/pull/1495. It should be available in the next `2.16.0` release.
